storage/innobase/row/row0merge.cc

```diff
@@ -2286,6 +2303,48 @@ row_merge_read_clustered_index(
 			}
 		}
 
+		if (old_table->versioned()) {
+			if (new_table->versioned() && !drop_historical) {
+				dfield_t *end = dtuple_get_nth_field(
+					row, new_table->vers_end);
+				byte *data = static_cast<byte *>(
+					dfield_get_data(end));
+				ut_ad(data);
+				if (mach_read_from_8(data) == TRX_ID_MAX) {
```

Comment:
I would suggest using a special predicate that does `memcmp()` against a string of 8 `0xff` bytes.